dask/dataframe/io.py

```diff
@@ -626,32 +626,55 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
 option to ensure that your dataset can be parallelized"""
 
-@wraps(pd.read_hdf)
-def read_hdf(path_or_buf, key, start=0, stop=None, columns=None,
-        chunksize=1000000):
-    with pd.HDFStore(path_or_buf) as hdf:
-        storer = hdf.get_storer(key)
-        if storer.format_type != 'table':
-            raise TypeError(dont_use_fixed_error_message)
-        if stop is None:
-            stop = storer.nrows
 
-    if columns is None:
-        columns = list(pd.read_hdf(path_or_buf, key, stop=0).columns)
+def _read_single_hdf(path, key, start=0, stop=None, columns=None,
+                     chunksize=int(1e6)):
+    def get_keys_and_stops(path, key, stop):
+        with pd.HDFStore(path) as hdf:
+            keys = [k for k in hdf.keys() if fnmatch(k, key)]
+            stops = []
+            for k in keys:
+                storer = hdf.get_storer(k)
+                if storer.format_type != 'table':
+                    raise TypeError(dont_use_fixed_error_message)
+                if stop is None:
+                    stops.append(storer.nrows)
+                else:
+                    stops.append(stop)
+        return keys, stops
+
+    def one_path_one_key(path, key, start, stop, columns, chunksize):
+        if columns is None:
+            columns = list(pd.read_hdf(path, key, stop=0).columns)
 
-    token = tokenize((path_or_buf, os.path.getmtime(path_or_buf), key, start,
-                      stop, columns, chunksize))
-    name = 'read-hdf-' + token
+        token = tokenize((path, os.path.getmtime(path), key, start,
+                          stop, columns, chunksize))
+        name = 'read-hdf-' + token
 
-    dsk = dict(((name, i), (apply, pd.read_hdf,
-                             (path_or_buf, key),
-                             {'start': s, 'stop': s + chunksize,
-                              'columns': columns}))
-                for i, s in enumerate(range(start, stop, chunksize)))
+        dsk = dict(((name, i), (apply, pd.read_hdf,
+                                 (path, key),
+                                 {'start': s, 'stop': s + chunksize,
+                                  'columns': columns}))
+                    for i, s in enumerate(range(start, stop, chunksize)))
 
-    divisions = [None] * (len(dsk) + 1)
+        divisions = [None] * (len(dsk) + 1)
+        return DataFrame(dsk, name, columns, divisions)
 
-    return DataFrame(dsk, name, columns, divisions)
+    keys, stops = get_keys_and_stops(path, key, stop)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize)
+                   for k, s in zip(keys, stops)])
+
+
+@wraps(pd.read_hdf)
+def read_hdf(pattern, key, start=0, stop=None, columns=None,
+             chunksize=1000000):
+    paths = sorted(glob(pattern))
+    if (start != 0 or stop is not None) and len(paths) > 1:
+        raise NotImplementedError("start and stop are ambiguous for more than "
+                                  "one file.")
```

Comment:
To a new user it's not clear what ambiguous would mean.  They're probably pretty clear in what they mean by start/stop.